# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../clanlib06/directfb_fix.patch
# Copyright (C) 2004 - 2006 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./Sources/Display/Display/DirectFB/target_directfb.cpp.orig	2002-02-26 17:38:11.000000000 -0300
+++ ./Sources/Display/Display/DirectFB/target_directfb.cpp	2005-02-23 00:27:08.922333056 -0300
@@ -104,7 +104,7 @@
 
 unsigned int CL_Target_DirectFB::get_width() const
 {
-  unsigned int width = 0;
+  int width = 0;
 
   surface->GetSize (surface, &width, NULL);
 
@@ -113,7 +113,7 @@
 
 unsigned int CL_Target_DirectFB::get_height() const
 {
-  unsigned int height = 0;
+  int height = 0;
 
   surface->GetSize (surface, NULL, &height);
 
@@ -147,7 +147,7 @@
     {
     case DSPF_A8:
       return 0;
-    case DSPF_RGB15:
+    case DSPF_ARGB1555:
       return 0x007c00;
     case DSPF_RGB16:
       return 0x00f800;
@@ -170,7 +170,7 @@
     {
     case DSPF_A8:
       return 0;
-    case DSPF_RGB15:
+    case DSPF_ARGB1555:
       return 0x0003e0;
     case DSPF_RGB16:
       return 0x0007e0;
@@ -193,7 +193,7 @@
     {
     case DSPF_A8:
       return 0;
-    case DSPF_RGB15:
+    case DSPF_ARGB1555:
     case DSPF_RGB16:
       return 0x00001f;
     case DSPF_RGB24:
@@ -215,7 +215,7 @@
     {
     case DSPF_A8:
       return 0xff;
-    case DSPF_RGB15:
+    case DSPF_ARGB1555:
     case DSPF_RGB16:
     case DSPF_RGB24:
     case DSPF_RGB32:
@@ -263,7 +263,7 @@
 
 void CL_Target_DirectFB::Clear (float r, float g, float b, float a)
 {
-  unsigned int width, height;
+  int width, height;
 
   surface->GetSize (surface, &width, &height);
 
